code/game/objects/items/weapons/power_cells.dm

```diff
@@ -467,14 +467,14 @@
 	suitable_cell = /obj/item/cell
 
 /obj/item/device/manual_charger/attack_self(mob/user)
-	var/obj/item/cell/cell
-	if(do_after(user, 60 - (1 * user.stats.getMult(STAT_TGH, STAT_LEVEL_ADEPT))))
+	if(!cell)
+		return
+	user.visible_message(SPAN_NOTICE("[user] starts turning the handle on [src]."), SPAN_NOTICE("You start to turn the handle on [src]."))
+	if(do_after(user, 5 + (20 * user.stats.getMult(STAT_TGH, STAT_LEVEL_ADEPT))))
 		if(!cell)
 			return
 		if(cell.charge >= cell.maxcharge)
```

Comment:
```suggestion
		if(cell.charge >= cell.maxcharge && (cell.charge - cell.maxcharg <= 20))
```
This could be not needed but it should fix any issues with people cheating it to over fill a cell